### ISSUE TITLE
arc4random_addrandom() not exported

### ISSUE THREAD

**Issue by pprindeville** [libevent/libevent#615]
`evutil_random.c:198` (in 2.1.8) references `arc4random_addrandom` as:

```
void
evutil_secure_rng_add_bytes(const char *buf, size_t n)
{
        arc4random_addrandom((unsigned char*)buf,
            n>(size_t)INT_MAX ? INT_MAX : (int)n);
}
```

but `arc4random_addrandom()` isn't exported by any header, causing:

```
[2018-03-28 15:41:49] build/libevent/libevent-2.1.8-stable/evutil_rand.c: In function 'evutil_secure_rng_add_bytes':
[2018-03-28 15:41:49] build/libevent/libevent-2.1.8-stable/evutil_rand.c:198:2: warning: implicit declaration of function 'arc4random_addrandom' [-Wimplicit-function-declaration]
[2018-03-28 15:41:49]   arc4random_addrandom((unsigned char*)buf,
[2018-03-28 15:41:49]   ^
[2018-03-28 15:41:49] build/libevent/libevent-2.1.8-stable/evutil_rand.c:198:2: warning: nested extern declaration of 'arc4random_addrandom' [-Wnested-externs]
```

when building.

**Comment by pprindeville** [comment]
No, wait... `./arc4random.c` gets included directly into `evutil_rand.c` so something else is stopping it from building on Android.

**Comment by azat** [comment]
Do you have this patch 266f43af7798befa3d27bfabaa9ae699259c3924?

**Comment by vincentbernat** [comment]
A release of libevent would be welcome to fix this issue. As the `arc4random_addrandom()` is being deprecated by some platforms, compilation of libevent fails. On Android, the function doesn't exist anymore. On macOS, it has been deprecated.

**Comment by azat** [comment]
@pprindeville does mentioned patch fixes the issue for you?

**Comment by siepkes** [comment]
On Illumos / SmartOS `arc4random_addrandom` is also not present (this [issue](https://www.illumos.org/issues/5949)).

**Comment by delphij** [comment]
FreeBSD will remove the arc4random_addrandom / arc4random_restir interfaces soon too.

See: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=230756 for progress.

**Comment by delphij** [comment]
Our proposed patch at: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=230764 (https://bz-attachments.freebsd.org/attachment.cgi?id=196373)

**Comment by delphij** [comment]
It seems that this can be fixed by cherry-picking 6541168d7 to the 2.1 branch (which will apply cleanly).

**Comment by azat** [comment]
2.1.9-beta released in 9c75c24, closing

**Comment by enjoysmath** [comment]
Same f*cking shit on the Raspberry Pi. Now what do we do?
